test/state/state_view.hpp

```diff
@@ -0,0 +1,50 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2024 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <evmc/evmc.hpp>
+#include <intx/intx.hpp>
+#include <optional>
+#include <unordered_map>
+#include <unordered_set>
+
+namespace evmone::state
+{
+using evmc::address;
+using evmc::bytes;
+using evmc::bytes32;
+using intx::uint256;
+
+class StateView
+{
+public:
+    struct Account
+    {
+        uint64_t nonce = 0;
+        uint256 balance;
+        bytes32 code_hash;
+        bool has_storage = false;
+    };
+
+    virtual ~StateView() = default;
+    [[nodiscard]] virtual std::optional<Account> get_account(address addr) const noexcept = 0;
+    [[nodiscard]] virtual bytes get_account_code(address addr) const noexcept = 0;
+    [[nodiscard]] virtual bytes32 get_storage(address addr, bytes32 key) const noexcept = 0;
+};
+
+struct StateDiff
```

Comment:
This layout doesn't look great. Single map of accounts should be better staring point?